src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -986,6 +1010,36 @@ impl LinkCollector<'_, '_> {
                 }
             }
 
+            // Strip generics from the path.
+            if path_str.contains(['<', '>'].as_slice()) {
+                stripped_path_string = match strip_generics_from_path(path_str) {
+                    Ok(path) => path,
+                    Err(err_kind) => {
+                        debug!("link has malformed generics: {}", path_str);
+                        resolution_failure(
```

Comment:
I don't think this should be a resolution failure, this should be an early return. If the path happens to have a single angle bracket, we ignore it.

Reply:
But then that won't catch mistakes like `Vec<T`. Are you suggesting this so that we don't interpret `a < b` as malformed generics?

Reply:
@Manishearth this is waiting on your review.

Personally I like catching `Box<Vec<T>` (which I think is more likely to pop up). Silently ignoring some links has always seemed really uninituitive to me, you can always use `[a < b]` in backticks if it conflicts with something that not a link.

Reply:
Note that `[a < b]` will only cause an error if the link is not resolved in the Markdown (e.g. it will work if you have `[a < b](rust-lang.org)`). See the test cases I added in 545f56104f2d81a6456ae0c645e6e310d913d3cc.

Reply:
@Manishearth this is waiting on your review :)

I think we're almost there; this is the last thing to resolve!

Reply:
Resolved in https://github.com/rust-lang/rust/pull/76934#pullrequestreview-506036317 :)